### PR TITLE
Added initial intel sklearnex support

### PR DESCRIPTION
**Pull request by PivovarA**
## Added initial intel sklearnex support

Hi everyone!
I would like to add initial **intel sklearex** support to **pycaret**. The **Intel extension for sklearn** is a set of optimized machine learning algorithms that have a sklearn-like API. More details can be viewed [here](https://github.com/intel/scikit-learn-intelex).
Integration of **intel extension for sklearn** into **pycaret** has already been discussed in this [issue](https://github.com/pycaret/pycaret/issues/1971).

## Description of changes

I have added the use_intelex flag to the _classification_, _clustering_, _regression_ and _anomaly_ files. I added patching of some algorithms to the _internal/tabular_ file in the setup method.
I didn't patch all the algorithms because, as you can see from this [issue](https://github.com/intel/scikit-learn-intelex/issues/996), some sklearnex algorithms can have not quite aligned with sklearn behavior. I added patching of algorithms that are most aligned with sklearn in terms of accuracy.
Also I have added tests and mentions of **sklearnex** in **pycaret** documentation

## Type of change

- [X] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

I used _compare_models_ and _create_model_ methods. Some algorithms have been changed to those optimized from **daal4py**. I have added the relevant tests as well. I have tested that everything works correctly with and without the _use_intelex_ flag. 

image

#### Describe if there is any unusual behaviour of your code(Write NA if there isn't)

NA

## Checklist:

- [X] My code follows the style guidelines of this project.
- [X] I have performed a self-review of my own code.
- [X] I have commented my code, particularly in hard-to-understand areas.
- [X] I have made corresponding changes to the documentation.
- [X] My changes generate no new warnings.
- [X] I have added tests that prove my fix is effective or that my feature works.
- [X] New and existing unit tests pass locally with my changes.
- [X] Any dependent changes have been merged and published in downstream modules.

## Difficulties

I'm not sure about the description of the _use_intelex_ flag and the description of **intel sklearnex** in the documentation. I didn't find any mentions of how to do this for **pycaret**, so I used text that is similar to the description of the _use_gpu_ flag.

### CODE REVIEW

**Comment by ngupta23** [comment]
@PivovarA We need to submit this to develop branch right now. Can you check the conflicts please? Thanks!

**Comment by PivovarA** [comment]
> @PivovarA We need to submit this to develop branch right now. Can you check the conflicts please? Thanks!

Hi
I can fix this, but as I see it, the code in the develop branch is organized a bit differently. Therefore, the fix may take some time

**Comment by ngupta23** [comment]
@Yard1 on second thought, I would agree. That would be the preferred approach to support alternate imports. So are we good with merging this PR then - https://github.com/pycaret/pycaret/pull/2552?

@PivovarA If this get's approved and merged, we should use the approach described in https://github.com/pycaret/pycaret/issues/2323 for this. So you would have something like the following to switch the models

```python
create_model("lr", engine="sklearn")
create_model("lr", engine="sklearnex") 
```

**Comment by Yard1** [comment]
Can we have different default engines for different models?

**Comment by ngupta23** [comment]
> Can we have different default engines for different models?

We will have to work through the logistics of this and various use cases. Shall we meet on Saturday (regular meeting) to discuss this? @PivovarA Let me know if you would like to join in, I can forward you the invite.

**Comment by PivovarA** [comment]
> > Can we have different default engines for different models?
> 
> We will have to work through the logistics of this and various use cases. Shall we meet on Saturday (regular meeting) to discuss this? @PivovarA Let me know if you would like to join in, I can forward you the invite.

Yes, that would be great

**Comment by ngupta23** [comment]
> Yes, that would be great

Can you join our slack channel and send me a pm with your email. I will send you the invite. Thanks.

**Comment by ngupta23** [comment]
@PivovarA The engines implementation is done now https://github.com/pycaret/pycaret/pull/2552 so you can make your changes in a similar manner

**Comment by PivovarA** [comment]
> @PivovarA The engines implementation is done now #2552 so you can make your changes in a similar manner

Hi @ngupta23 and @Yard1 
I saw your changes. 
The code is already in my [repository](https://github.com/PivovarA/pycaret/tree/intelex_develop) and I can create a PR at any time. 
However, I found another issue from our side (https://github.com/intel/scikit-learn-intelex/issues/1007). This is because scikit-learn has released the latest version but intel extension haven't. This bug has already been fixed, but due to the fact that we have delayed the release of the package, we can now observe it. I can create a PR, however, the sklearnex engine will only work with a slightly earlier version of sklearn (1.0.2). Therefore, I would like to ask: is it worth creating a PR now?

**Comment by ngupta23** [comment]
@Yard1 @PivovarA , the hard limit is as follows, so should it not be ok for you. If they want to use intel accelerators, they should still be able to install it in the same environment - correct?

scikit-learn>=1.0

**Comment by PivovarA** [comment]
Hi @ngupta23 and @Yard1 
It seems that problem was resolved for python 3.10.
Can I still open a PR for the develop branch?

**Comment by ngupta23** [comment]
Yes. We are about to make a release in the coming week. If you can make the  PR in the next couple of days, then go ahead with the develop branch. Thanks

**Comment by PivovarA** [comment]
Hi @ngupta23  and @Yard1. 
I tried to run tests with intel extension for _sklearn_ for **python 3.10** and realized that there are some libraries in _pycaret_ that don't support python 3.10 (_sktime_). 
As I can see on their pipy page, they only support versions up to **python 3.9**. While _sklearnex_ requires **python 3.10**. As @Yard1 said, I want intel extension and pycaret to be fully compatible. I think that this problem will be resolved by the next release of our library. At least I will push the team to resolve this issue. Is it okay?

**Comment by Yard1** [comment]
That sounds fine to me! We'll make sure that sklearnex goes into final 3.0 release.

**Comment by moezali1** [comment]
@Yard1 Should we test this in the next rc release instead of final release? Let me know what you think?

**Comment by Yard1** [comment]
Yeah, we should try to fit it in the RC

**Comment by moezali1** [comment]
@PivovarA Do you have any updates on this?

**Comment by PivovarA** [comment]
> @PivovarA Do you have any updates on this?

Hi @ngupta23, @Yard1 and @moezali1 
I would like to say that we fixed an issue with incorrect metadata for intel extension for scikit-learn. Now it can be installed with python version >= 3.7. 
How much time do I have to finish this PR? I'm planning to deal with it this week. Is it ok?

**Comment by Yard1** [comment]
That is fine! We still need to finish some things on our end
